hazelcast/src/hazelcast/client/serialization/pimpl/SerializationConstants.cpp

```diff
@@ -18,12 +18,15 @@
 //
 
 #include "hazelcast/client/serialization/pimpl/SerializationConstants.h"
+#include "hazelcast/util/Util.h"
+#include "hazelcast/util/ILogger.h"
+#include "hazelcast/client/exception/IClassCastException.h"
 
 namespace hazelcast {
     namespace client {
         namespace serialization {
             namespace pimpl {
-                SerializationConstants *SerializationConstants::instance = NULL;
+                SerializationConstants *SerializationConstants::instance = &getInstance();
```

Comment:
I was thinking `SerializationConstants`, because constructor is private.But I just tried and I can call the constructor here. It looks like this part counts as inside of SerializationConstants scope. 
And I was taking this link as reference
http://stackoverflow.com/questions/17924688/meyers-singleton-thread-safe-with-c-98

Now reading carefully, the discussion about thread safety is not clear. We may have to use locks here. And double checking lock guarantee is not available in c++98. I can reason that new implementation in this prd safer, but I am not sure if this is safe. See following,
http://www.aristeia.com/Papers/DDJ_Jul_Aug_2004_revised.pdf

Or better way, remove the singleton approach completely.